src/main/java/org/apache/sling/commons/scheduler/impl/QuartzThreadPool.java

```
@@ -21,13 +21,18 @@
 public class QuartzThreadPool implements org.quartz.spi.ThreadPool {
 
     /** Our executor thread pool */
-    private ThreadPool executor;
+    private volatile ThreadPool executor;
+
+    private final Object lock = new Object();
+
+    private volatile int counter;
 
     /**
      * Create a new wrapper implementation for Quartz.
      */
     public QuartzThreadPool(final ThreadPool executor) {
         this.executor = executor;
+        this.counter = executor.getConfiguration().getMaxPoolSize();
```

can `maxPoolSize` ever be zero or negative?

---

Looking at the code in Sling commons thread pool, it seems it can :( we should maybe just throw an exception in this case. And eventually fix commons thread pool to not return negative values (we might need to check other config values as well)

---

actually on second thought : if it's negative DefaultThreadPool constructor converts it to Integer.MAX_VALUE and if it's zero ThreadExpiringThreadPool constructor will fail with an exception. so we might actually be fine.